otj-metrics-core/src/main/java/com/opentable/metrics/graphite/OtGraphiteReporter.java

```diff
@@ -0,0 +1,207 @@
+        graphite.send(prefix(name, "p999"), format(convertDuration(snapshot.get999thPercentile())), timestamp);
+        reportMetered(name, timer, timestamp);
+    }
+
+    private void reportMetered(String name, Metered meter, long timestamp) throws IOException {
+        this.reportCounter(name, meter.getCount(), timestamp);
+        graphite.send(prefix(name, "m1_rate"), format(convertRate(meter.getOneMinuteRate())), timestamp);
+        graphite.send(prefix(name, "m5_rate"), format(convertRate(meter.getFiveMinuteRate())), timestamp);
+        graphite.send(prefix(name, "m15_rate"), format(convertRate(meter.getFifteenMinuteRate())), timestamp);
+        graphite.send(prefix(name, "mean_rate"), format(convertRate(meter.getMeanRate())), timestamp);
+    }
+
+    private void reportHistogram(String name, Histogram histogram, long timestamp) throws IOException {
+        final Snapshot snapshot = histogram.getSnapshot();
+        this.reportCounter(name, histogram.getCount(), timestamp);
+        graphite.send(prefix(name, "max"), format(snapshot.getMax()), timestamp);
+        graphite.send(prefix(name, "mean"), format(snapshot.getMean()), timestamp);
+        graphite.send(prefix(name, "min"), format(snapshot.getMin()), timestamp);
+        graphite.send(prefix(name, "stddev"), format(snapshot.getStdDev()), timestamp);
+        graphite.send(prefix(name, "p50"), format(snapshot.getMedian()), timestamp);
+        graphite.send(prefix(name, "p75"), format(snapshot.get75thPercentile()), timestamp);
+        graphite.send(prefix(name, "p95"), format(snapshot.get95thPercentile()), timestamp);
+        graphite.send(prefix(name, "p98"), format(snapshot.get98thPercentile()), timestamp);
+        graphite.send(prefix(name, "p99"), format(snapshot.get99thPercentile()), timestamp);
+        graphite.send(prefix(name, "p999"), format(snapshot.get999thPercentile()), timestamp);
+    }
+
+    private void reportCounter(String name, Counter counter, long timestamp) throws IOException {
+        this.reportCounter(name, counter.getCount(), timestamp);
+    }
+
+    private void reportCounter(String name, long value, long timestamp) throws IOException {
+        graphite.send(prefix(name, "count"), format(value), timestamp);
+        final long diff = value - Optional.ofNullable(reportedCounters.put(name, value)).orElse(0L);
+        if (diff != 0L) {
+            graphite.send(prefix(name, "hits"), format(diff), timestamp);
+            graphite.send(prefix(name, "cps"), format(diff * countFactor), timestamp);
+        }
+    }
+
+    private void reportGauge(String name, Gauge gauge, long timestamp) throws IOException {
+        final String value = format(gauge.getValue());
+        if (value != null) {
+            graphite.send(prefix(name), value, timestamp);
+        }
+    }
+
+    private String format(Object o) {
+        if (o instanceof Float) {
+            return format(((Float) o).doubleValue());
```

Comment:
You can just cast to Number and use https://docs.oracle.com/javase/7/docs/api/java/lang/Number.html#longValue() for most (if not all) of these.

Reply:
this is copy/paste of the older graphite reporter